### PR TITLE
feat(PopoverConfirm): new component

### LINKED ISSUE #382: [Feat]: new PopoverConfirm component
Description: Popover Confirm designed to request and capture user confirmation for actions in a visually prominent yet non-intrusive manner. It overlays the primary content to present a confirmation dialog that includes clear options for users to confirm or cancel the action.
Requirements:

Component Implementation:
Implement the Popover Confirm component in TypeScript.
Utilize SCSS with BEM methodology for styling.
Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, shadows, and animations.
Support various placements (e.g., top, bottom, left, right) and trigger types (e.g., click, hover, focus).
Ensure accessibility, including proper ARIA roles and keyboard navigation support.
Unit Tests:
Write unit tests using Jest and React Testing Library.
Ensure test coverage for all props, including rendering of different placements, trigger types, and interaction states.
Test for edge cases such as missing content or unsupported placement values.
Storybook Stories:
Create a Storybook story for the Popover Confirm component.
Showcase different placements (e.g., top, bottom, left, right) and trigger types (e.g., click, hover).
Create controls for the props so users can interact with the Popover Confirm component in Storybook.
Document all props of the public interfaces.

### Checklist:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from `@geneui/tokens` are used for colors, font sizes, spacing, shadows, and animations.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and placement types.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by HaykKirakosyanSoft**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1340 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-gkynnqsdqc.chromatic.com/

### CODE REVIEW

File: src/components/atoms/Popover/index.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
 export { IPopoverProps, IPopoverRef, default as Popover } from "./Popover";
 export { default as PopoverBody } from "./PopoverBody";
-export { default as PopoverFooter } from "./PopoverFooter";
-export { default as PopoverFooterActions } from "./PopoverFooterActions";
+export { default as PopoverFooter, IPopoverFooterProps, IPopoverFooterActionProps } from "./PopoverFooter";
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/index.tsx]: Please remove the separate export of IPopoverFooterActionProps unless it is used externally as a standalone public type; IPopoverFooterProps already covers this.

File: src/components/atoms/Popover/Popover.stories.tsx [hunk 2]
```diff
@@ -95,7 +107,7 @@ const WithoutFooterComponent: FC<IPopoverProps> = (props) => {
     const [propsForContent, setPropsForContent] = useState({});
     return (
         <div style={{ margin: "500px 500px", height: 1000 }}>
-            <Popover {...props} setProps={setPropsForContent}>
+            <Popover {...props} setProps={setPropsForContent} Icon={Info}>
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/Popover.stories.tsx]: Please remove `Icon={Info}` and pass it from args
```
    args: {
        Icon: Info
    }
```

File: src/components/atoms/Popover/Popover.stories.tsx [hunk 3]
```diff
@@ -95,7 +107,7 @@ const WithoutFooterComponent: FC<IPopoverProps> = (props) => {
...
                 <PopoverBody>
                     <div className="swapComponent" style={{ minHeight: "100%", background: "#F4E1EC" }} />
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/Popover.stories.tsx]: Please remove `style={{ minHeight: "100%", background: "#F4E1EC" }}` and pass some content like lorem ipsum

File: src/components/atoms/Popover/Popover.stories.tsx [hunk 4]
```diff
@@ -119,20 +131,24 @@ const WithoutHeaderComponent: FC<IPopoverProps> = (props) => {
...
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/Popover.stories.tsx]: Please remove style={{ minHeight: "100%", background: "#F4E1EC" }} and pass some content like lorem ipsum

File: src/components/atoms/Popover/Popover.stories.tsx [hunk 5]
```diff
@@ -119,20 +131,24 @@ const WithoutHeaderComponent: FC<IPopoverProps> = (props) => {
...
+                            onClick: () => {}
+                        },
+                        {
+                            text: "Secondary",
+                            appearance: "primary",
+                            onClick: () => {}
+                        }
+                    ]}
+                >
                     <div
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/Popover.stories.tsx]: Please remove the placeholder` <div className="swapComponent" style={{ minHeight: "100%", background: "#F4E1EC" }} /> ` and use an actual component example instead (e.g., `Avatar`, `Button`, etc.) to keep the story/demo aligned with real usage.

File: src/components/atoms/Popover/Popover.tsx [hunk 6]
```diff
@@ -186,6 +187,16 @@ export interface IPopoverProps {
      * Can be either "click" or "hover".
      */
     trigger?: "click" | "hover";
+    /**
+     * Determines whether the close (X) button is displayed in the popover's header.
+     * @default true
+     */
+    hasCloseButton?: boolean;
+    /**
+     * Popover header icon <br/>
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/Popover.tsx]: The current JSDoc is a bit misleading. This prop is used to render an icon in the popover header (before the title), not as an avatar. Please update the description accordingly.

File: src/components/atoms/Popover/PopoverFooter.tsx [hunk 7]
```diff
@@ -1,7 +1,96 @@
...
-    return <div className="popover__footer">{children}</div>;
+import { IconProps } from "@geneui/icons";
+
+// Components
+import Button from "@components/atoms/Button";
+import ButtonGroup from "@components/molecules/ButtonGroup";
+
+export interface IPopoverFooterActionProps {
+    /**
+     * The `Icon` component to display in the action button. If not provided, the action button will not be rendered.
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/PopoverFooter.tsx]: Please fix the description, `If not provided, the action button will not be rendered` this is wrong.

File: src/components/atoms/Popover/PopoverFooter.tsx [hunk 8]
```diff
@@ -1,7 +1,96 @@
...
+    /**
+     * The swappable content displayed in the footer.
+     */
+    children?: ReactNode;
+    /**
+     * An array of action button objects to display in the footer.
+     * The rendered buttons are automatically wrapped in a `ButtonGroup` component to ensure proper spacing and alignment.
+     * @example
+     * actions={[
+     *   { Icon: Globe, text: "View", onClick: handleAction },
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/PopoverFooter.tsx]: Can we provide one complete example object that includes all IPopoverFooterActionProps fields? The current example (`{ Icon: Globe, text: "View", onClick: handleAction }`) is helpful, but a fully populated sample would make the contract clearer for consumers.
`{
  text: "View",
  onClick: handleAction,
  Icon: Globe,
  variant: "primary",
  disabled: false,
  loading: false
}`

File: src/components/atoms/Popover/PopoverFooter.tsx [hunk 9]
```diff
@@ -1,7 +1,96 @@
...
+            ...action,
+            id: action.id || `popover-footer-action-${nanoid()}`
+        }));
+    }, [actions]);
+
+    return (
+        <div className="popover__footer">
+            {children}
+            {hasActions && (
+                <ButtonGroup className="popover__footer_buttons" size="medium">
```
Comment (GaroGabrielyan) [inline src/components/atoms/Popover/PopoverFooter.tsx]: Please rename `popover__footer_buttons` to a valid BEM-style semantic name (e.g., `popover__footerActions`). Current naming mixes separators and is inconsistent.

File: src/components/molecules/PopoverConfirm/PopoverConfirm.scss [hunk 10]
```diff
@@ -0,0 +1,9 @@
+.popoverConfirm__title_icon {
```
Comment (GaroGabrielyan) [inline src/components/molecules/PopoverConfirm/PopoverConfirm.scss]: Please rename `popoverConfirm__title_icon` to a valid BEM-style semantic name.

File: src/components/molecules/PopoverConfirm/PopoverConfirm.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    children,
+    primaryButtonText = "Confirm",
+    secondaryButtonText = "Cancel",
+    onConfirm,
+    onCancel,
+    onOpenChange,
+    open: controlledOpen,
+    defaultOpen = false,
+    status = "warning",
+    ...popoverProps
```
Comment (GaroGabrielyan) [inline src/components/molecules/PopoverConfirm/PopoverConfirm.tsx]: 🚨 I think spreading all Popover props into PopoverConfirm makes this component too open and weakens its contract. PopoverConfirm is a specific UX pattern and should expose only the props it actually supports by design. For example, allowed sizes seem limited (small/medium), margin should be internal constant, hasCloseButton appears fixed (false), and options like fitReference are not part of the current design. Could we remove broad prop spreading and define an explicit, constrained PopoverConfirm API instead?

File: src/components/molecules/PopoverConfirm/PopoverConfirm.tsx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+    const headerIcon: FC<IconProps> = ({ className, ...props }: IconProps) => (
+        <IconComponent {...props} className={classNames(className, `popoverConfirm__title_icon_${status}`)} size={20} />
+    );
+
+    return (
+        <div className="popoverConfirm">
+            <Popover {...popoverProps} ref={popoverRef} open={isOpenState} hasCloseButton={false} Icon={headerIcon}>
+                <PopoverBody>{children}</PopoverBody>
+                <PopoverFooter
+                    actions={[
```
Comment (GaroGabrielyan) [inline src/components/molecules/PopoverConfirm/PopoverConfirm.tsx]: Current footer action construction is too rigid: it always renders both secondary and primary actions. This makes cases like “primary-only confirm” impossible. Please make action generation conditional (or accept explicit actions) so consumers can render only the buttons they need.

File: src/components/molecules/PopoverConfirm/PopoverConfirm.tsx [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+                setIsOpenState(false);
+                onOpenChange?.(false);
+            }
+        },
+        [popoverRef.current.floatingElement]
+    );
+
+    const primaryButtonAppearance = status === "error" ? "danger" : "primary";
+
+    const IconComponent = status === "error" ? ErrorFilled : TriangleAlert;
```
Comment (GaroGabrielyan) [inline src/components/molecules/PopoverConfirm/PopoverConfirm.tsx]: Could we use a status-to-icon map instead of a ternary for better scalability/readability? Example: 
```
const iconByStatus = { error: ErrorFilled, warning: TriangleAlert }; 
const IconComponent = iconByStatus[status];
```

File: src/components/molecules/PopoverConfirm/PopoverConfirm.tsx [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+            if (!onReferenceClick && isOpenState) {
+                setIsOpenState(false);
+                onOpenChange?.(false);
+            }
+        },
+        [popoverRef.current.floatingElement]
+    );
+
+    const primaryButtonAppearance = status === "error" ? "danger" : "primary";
```
Comment (GaroGabrielyan) [inline src/components/molecules/PopoverConfirm/PopoverConfirm.tsx]: Could we align this with the same mapping approach to keep status-driven values consistent? Example: 
```
const appearanceByStatus = { error: "danger", warning: "primary" }; 
const primaryButtonAppearance = appearanceByStatus[status];
```

**Review by GaroGabrielyan — CHANGES_REQUESTED** [review]

**Review by GaroGabrielyan — APPROVED** [review]
